mmseg/models/decode_heads/dpt.py

```diff
@@ -0,0 +1,199 @@
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import (Conv2d, ConvModule, ConvTranspose2d,
+                      build_activation_layer, build_norm_layer)
+
+from ..builder import HEADS
+from ..utils import Transpose, _make_readout_ops
+from .decode_head import BaseDecodeHead
+
+
+class ViTPostProcessBlock(nn.Module):
+
+    def __init__(self,
+                 in_channels=768,
+                 out_channels=[96, 192, 384, 768],
+                 img_size=[384, 384],
+                 readout_type='ignore',
+                 start_index=1,
+                 kernel_sizes=[4, 2, 1, 3],
+                 strides=[4, 2, 1, 2],
+                 paddings=[0, 0, 0, 1]):
+        super(ViTPostProcessBlock, self).__init__()
+
+        self.readout_ops = _make_readout_ops(in_channels, out_channels,
+                                             readout_type, start_index)
+
+        self.unflatten_size = torch.Size(
+            [img_size[0] // 16, img_size[1] // 16])
+
+        self.post_process_ops = []
+        for idx, out_channels in enumerate(out_channels):
+            self.post_process_ops.append(
+                nn.Sequential(
+                    self.readout_ops[idx], Transpose(1, 2),
+                    nn.Unflatten(2, self.unflatten_size),
+                    Conv2d(in_channels, out_channels, kernel_size=1),
+                    ConvTranspose2d(
+                        out_channels,
+                        out_channels,
+                        kernel_size=kernel_sizes[idx],
+                        stride=strides[idx],
+                        padding=paddings[idx])))
+
+    def forward(self, inputs):
+        assert len(inputs) == len(self.readout_ops)
+        for idx, x in enumerate(inputs):
+            inputs[idx] = self.post_process_ops[idx](x)
+        return inputs
+
+
+class ResidualConvUnit(nn.Module):
```

Comment:
Can we add [PreActBottleneck](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/resnetv2.py#L121) and PreActBasicBlock according to [Bottleneck](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/backbones/resnet.py#L98) and [BasicBlock](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/backbones/resnet.py#L13)